st2common/st2common/services/workflows.py

```diff
@@ -1561,3 +1564,38 @@ def identify_orphaned_workflows():
             continue
 
     return orphaned
+
+
+def add_system_info_to_action_context(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        engine_info = system_info.get_process_info()
+
+        # Argument will be of type WorkflowExecutionDB/ActionExecutionDB only.
+        if kwargs.get("wf_ex_db"):
+            ac_ex = ex_db_access.ActionExecution.get_by_id(
+                kwargs["wf_ex_db"].action_execution
+            )
+            lv_ex = lv_db_access.LiveAction.get_by_id(ac_ex.liveaction["id"])
+        elif kwargs.get("ac_ex_db"):
+            lv_ex = lv_db_access.LiveAction.get_by_id(
+                kwargs["ac_ex_db"].liveaction["id"]
+            )
+        else:
+            raise ValueError("Invalid message type")
+
+        lv_ex.context.update({"engine_info": engine_info})
+        lv_db_access.LiveAction.add_or_update(lv_ex, publish=False)
+
+        try:
+            return func(*args, **kwargs)
+        except Exception as e:
+            raise e
+        finally:
+            lv_ex = lv_db_access.LiveAction.get_by_id(lv_ex.id)
```

Comment:
It's not clear why we need this code path so we should add a comment on why this delete is here and why it's needed.

Reply:
@Kami The intention here is to add engine information to the context at the start of handling and remove it once handling is complete.
Any action in running/pausing/cancelling state, having engine information about a deregistered service can be classified as orphaned workflow.

Reply:
Since the workflow execution can be processed by 1 or more different workflow engine during its execution, it's difficult to know which workflow engine is currently handling the execution.